dbms/src/Operators/AggregateContext.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Common/Logger.h>
+#include <Interpreters/Aggregator.h>
+#include <Operators/Operator.h>
+
+namespace DB
+{
+
+struct ThreadData
+{
+    size_t src_rows = 0;
+    size_t src_bytes = 0;
+
+    ColumnRawPtrs key_columns;
+    Aggregator::AggregateColumns aggregate_columns;
+
+    ThreadData(size_t keys_size, size_t aggregates_size)
+    {
+        key_columns.resize(keys_size);
+        aggregate_columns.resize(aggregates_size);
+    }
+};
+
+class AggregateContext
+{
+public:
+    AggregateContext(
+        bool final_,
+        const String & req_id)
+        : log(Logger::get(req_id))
+        , is_final(final_)
+    {
+    }
+
+    void init(const Aggregator::Params & params, size_t max_threads_);
```

Comment:
how about `initBuild`
```suggestion
    void initBuild(const Aggregator::Params & params, size_t max_threads_);
```